clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -338,6 +383,175 @@ impl FunctionType {
     }
 }
 
+pub fn trait_check_trait_compliance(
+    db: &mut AnalysisDatabase,
+    contract_context: Option<&ContractContext>,
+    actual_trait_identifier: &TraitIdentifier,
+    actual_trait: &BTreeMap<ClarityName, FunctionSignature>,
+    expected_trait_identifier: &TraitIdentifier,
+    expected_trait: &BTreeMap<ClarityName, FunctionSignature>,
+) -> CheckResult<()> {
+    // Shortcut for the simple case when the two traits are the same.
+    if actual_trait_identifier == expected_trait_identifier {
+        return Ok(());
+    }
+
+    for (func_name, expected_sig) in expected_trait.iter() {
```

Comment:
So, `actual_trait` could contain functions that are not in `expected_trait`?  Is this permitted?

Reply:
No, in Clarity1 when passing a trait to a trait, the types must match exactly. In Clarity2, we allow this concept of "compatible traits".

Reply:
can the method comment reflect this?

Reply:
Added in d0327625a